examples/gcoap_block_server/gcoap_block.c

```diff
@@ -0,0 +1,221 @@
+            (unsigned)block1.offset, pdu->payload_len, blockwise, block1.more);
+
+    /* initialize sha256 calculation and add payload bytes */
+    if (block1.blknum == 0) {
+        puts("_sha256_handler: init");
+        sha256_init(&sha256);
+    }
+    sha256_update(&sha256, pdu->payload, pdu->payload_len);
+
+    unsigned resp_code = COAP_CODE_CHANGED;
+    if (block1.more) {
+        resp_code = COAP_CODE_CONTINUE;
+    }
+
+    /* start response */
+    gcoap_resp_init(pdu, buf, len, resp_code);
+
+    /* has payload */
+    if (!blockwise || !block1.more) {
+        coap_opt_add_format(pdu, COAP_FORMAT_TEXT);
+    }
+    if (blockwise) {
+        coap_opt_add_block1_control(pdu, &block1);
+    }
+
+    /* include digest if done, otherwise response code above asks for next block */
+    size_t pdu_len = 0;
+    if (!blockwise || !block1.more) {
+        puts("_sha256_handler: finish");
+        sha256_final(&sha256, digest);
+        pdu_len = coap_opt_finish(pdu, COAP_OPT_FINISH_PAYLOAD);
+
+        pdu_len += fmt_bytes_hex((char *)pdu->payload, digest, sizeof(digest));
+    }
+    else {
+        pdu_len = coap_opt_finish(pdu, COAP_OPT_FINISH_NONE);
+    }
+
+    return pdu_len;
+}
+
+int gcoap_cli_cmd(int argc, char **argv)
+{
+    if (argc == 1) {
+        /* show help for main commands */
+        goto end;
+    }
+
+    if (strcmp(argv[1], "info") == 0) {
+        uint8_t open_reqs = gcoap_op_state();
+
+        printf("CoAP server is listening on port %u\n", CONFIG_GCOAP_PORT);
+        printf("CoAP open requests: %u\n", open_reqs);
+        return 0;
+    }
+
+    end:
+    printf("usage: %s <info>\n", argv[0]);
+    return 1;
+}
+
+void gcoap_cli_init(void)
+{
+#ifdef MODULE_SOCK_DTLS
+#ifdef DTLS_PSK
+    credman_credential_t credential = {
+        .type = CREDMAN_TYPE_PSK,
+        .tag = SOCK_DTLS_GCOAP_TAG,
+        .params = {
+            .psk = {
+                .key = { .s = (char *)psk_key, .len = psk_key_len },
+                .id = { .s = (char *)psk_id, .len = psk_id_len },
+            },
+        },
+    };
+#else
+    ecdsa_public_key_t other_pubkeys[] = {
+        { .x = ecdsa_pub_key_x, .y = ecdsa_pub_key_y },
+    };
+
+    credman_credential_t credential = {
+        .type = CREDMAN_TYPE_ECDSA,
+        .tag = SOCK_DTLS_GCOAP_TAG,
+        .params = {
+            .ecdsa = {
+                .private_key = ecdsa_priv_key,
+                .public_key = {
+                    .x = ecdsa_pub_key_x,
+                    .y = ecdsa_pub_key_y,
+                },
+                .client_keys = other_pubkeys,
+                .client_keys_size = ARRAY_SIZE(other_pubkeys),
+            }
+        },
+    };
+#endif /* DTLS_ECC */
+    if (credman_add(&credential) < 0) {
+        puts("gcoap_cli: unable to add credential");
+        return;
+    }
+
+    /* tell gcoap with tag to use */
+    gcoap_set_credential_tag(SOCK_DTLS_GCOAP_TAG);
+    /* start gcoap server */
+    gcoap_init();
+#endif
```

Comment:
since there is another example for DTLS we might want to remove that from this one

Reply:
Isn't the other example for a DTLS *client*?

Reply:
But this one seem to be a bit more flexible than the one in gcoap_dtls.

Reply:
> Isn't the other example for a DTLS _client_?

no it is server and client

Reply:
I removed DTLS code to keep things simple.